packages/gatsby/src/schema/types/sort.ts

```diff
@@ -65,6 +65,7 @@ const convert = <TContext = any>({
   const sortFields = {}
 
   Object.keys(fields).forEach(fieldName => {
+    let deprecationReason = parentFieldDeprecationReason
```

Comment:
Github doesn't show this as rest of the code is few lines below but this is because 

https://github.com/gatsbyjs/gatsby/blob/d846f8981cdf2dd866cec3d7d191b8f64bbbc2b3/packages/gatsby/src/schema/types/sort.ts#L67-L78

In particular on line 77 once a field is marked as deprecated, all the remaining fields in current iteration are also deprecated

For example:
```graphql
type Foo {
  field1: Int # this is fine
  field2(arg1: Int): Int # this sets `deprecationReason`
  field3: Int # this should be fine, but currently marked as deprecated
}
```

Reply:
Does this fix https://github.com/gatsbyjs/gatsby/issues/31523 ?

Reply:
Most likely, yes - at least comments about fields that don't have arguments but still causing this deprecation to show up

Reply:
I can move this to separate PR as this change is not dependend on rest of this PR, I did add it here as this warning was part of "issues" I encountered. The main problem for me was that Graphiql was not autosuggesting fields due to them being deprecated, but it was still working if I manually typed them in

Reply:
Can stay in imo, I was just curious :)